salt/modules/cp.py

```diff
@@ -357,9 +359,11 @@ def get_file_str(path, saltenv='base', env=None):
         saltenv = env
 
     fn_ = cache_file(path, saltenv)
-    with salt.utils.fopen(fn_, 'r') as fp_:
-        data = fp_.read()
-    return data
+    if isinstance(fn_, str):
```

Comment:
@rallytime Fixed, thanks!

`os.path.exists(fn_)` doesn't suit here, because it will throw `TypeError` if `cache_file()` function would return `False` (file cannot be cached for some reason). Too much logic to wrap around this simple check.